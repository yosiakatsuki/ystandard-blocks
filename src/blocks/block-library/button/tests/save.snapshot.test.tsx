import React from 'react';
import { render } from '@testing-library/react';
import Save from '../save';

// スナップショットテスト
describe( 'Button Block <Save /> snapshot', () => {
	it( 'matches snapshot with minimal attributes', () => {
		const { asFragment } = render(
			<Save
				attributes={ {
					content: 'Button',
					url: 'https://example.com',
				} }
			/>
		);
		expect( asFragment() ).toMatchSnapshot();
	} );

	it( 'matches snapshot with icons and all attributes', () => {
		const { asFragment } = render(
			<Save
				attributes={ {
					content: 'Button',
					url: 'https://example.com',
					iconLeft: 'left',
					iconRight: 'right',
					iconLeftBlank: true,
					iconRightBlank: true,
					iconSizeLeft: 16,
					iconSizeRight: 20,
					fontSize: 'large',
					customFontSize: 24,
					gradient: 'gradient',
					linkTarget: '_blank',
					rel: 'noopener',
				} }
			/>
		);
		expect( asFragment() ).toMatchSnapshot();
	} );

	it( 'matches snapshot with borderRadius 0', () => {
		const { asFragment } = render(
			<Save
				attributes={ {
					content: 'Button',
					url: 'https://example.com',
					borderRadius: '0px',
				} }
			/>
		);
		// スナップショットでHTML全体を確認
		expect( asFragment() ).toMatchSnapshot();
	} );
} );
