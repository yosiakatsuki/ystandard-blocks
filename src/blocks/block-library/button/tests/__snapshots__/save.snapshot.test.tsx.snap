// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Block <Save /> snapshot matches snapshot with borderRadius 0 1`] = `
<DocumentFragment>
  <div
    class="ystdb-custom-button"
  >
    <a
      class="ystdb-custom-button__link"
      href="https://example.com"
      style="border-radius: 0;"
    >
      <span
        class="ystdb-custom-button__content"
      >
        Button
      </span>
    </a>
  </div>
</DocumentFragment>
`;

exports[`Button Block <Save /> snapshot matches snapshot with icons and all attributes 1`] = `
<DocumentFragment>
  <div
    class="ystdb-custom-button"
  >
    <a
      class="ystdb-custom-button__link"
      href="https://example.com"
      rel="noopener"
      style="font-size: 24px;"
      target="_blank"
    >
      <span
        aria-hidden="true"
        class="ys-icon ys-icon--fallback"
        style="font-size: 16px;"
      >
        <svg
          xmlns="http://www.w3.org/2000/svg"
        />
      </span>
      <span
        class="ystdb-custom-button__content"
      >
        Button
      </span>
      <span
        aria-hidden="true"
        class="ys-icon ys-icon--fallback"
        style="font-size: 20px;"
      >
        <svg
          xmlns="http://www.w3.org/2000/svg"
        />
      </span>
    </a>
  </div>
</DocumentFragment>
`;

exports[`Button Block <Save /> snapshot matches snapshot with minimal attributes 1`] = `
<DocumentFragment>
  <div
    class="ystdb-custom-button"
  >
    <a
      class="ystdb-custom-button__link"
      href="https://example.com"
    >
      <span
        class="ystdb-custom-button__content"
      >
        Button
      </span>
    </a>
  </div>
</DocumentFragment>
`;
